Reduce build times by refactoring FrameView.h

jernoble opened the pull request:
#### 4c8702ace4ab5ce127a144a8339bfee8f5b48a24
<pre>
Reduce build times by refactoring FrameView.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=251626">https://bugs.webkit.org/show_bug.cgi?id=251626</a>
rdar://104975520

Reviewed by Brent Fulgham.

Forward declare more things in FrameView.h and remove #include statements.

In order to forward-declare some enumerations used in FrameView.h, those enums
needed to be moved out of their classes&apos; scopes, and in one case, need to be
converted to an OptionSet.

Canonical link: <a href="https://commits.webkit.org/259960@main">https://commits.webkit.org/259960@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d7a3415829d68abcbccb46075373ecc16e5c26c2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/106458 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/15482 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/39274 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/115645 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/175745 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/110366 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/16971 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/6707 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/98656 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/115309 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/112224 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/12891 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95865 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/40450 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/94766 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/27509 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/82163 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/8714 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28861 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/9247 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/5432 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/14868 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/48407 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6887 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/10795 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/platform/graphics/ca/TileController.cpp:
```diff
@@ -242,7 +242,7 @@ void TileController::setVelocity(const VelocityData& velocity)
 }
 
-void TileController::setScrollability(Scrollability scrollability)
+void TileController::setScrollability(OptionSet<Scrollability> scrollability)
```
An equivalent change is needed in `Source\WebKit\WebProcess\WebPage\wc\GraphicsLayerWC.cpp`

brentfulgham commented on Source/WebCore/page/FrameView.h:
```diff
@@ -27,21 +27,13 @@
-#include "ScrollView.h"
 #include "SimpleRange.h"
-#include "StyleColor.h"
-#include "TiledBacking.h"
```
Nice!

brentfulgham commented on Source/WebCore/page/FrameView.h:
```diff
@@ -27,21 +27,13 @@
 #include "Color.h"
-#include "ContainerNode.h"
 #include "FrameViewLayoutContext.h"
-#include "GraphicsContext.h"
```
Some Windows code seemed to expect this, maybe just `WebProcess\WebPage\CoordinatedGraphics\LayerTreeHostTextureMapper.cpp`.

brentfulgham commented on Source/WebCore/page/FrameView.h:
```diff
@@ -343,16 +347,16 @@ class FrameView final : public AbstractFrameView {
     void removeSlowRepaintObject(RenderElement&);
-    bool hasSlowRepaintObject(const RenderElement& renderer) const { return m_slowRepaintObjects && m_slowRepaintObjects->contains(renderer); }
-    bool hasSlowRepaintObjects() const { return m_slowRepaintObjects && !m_slowRepaintObjects->isEmptyIgnoringNullReferences(); }
+    bool hasSlowRepaintObject(const RenderElement& renderer) const;
```
Remove the `renderer` label.

brentfulgham reviewed (approved):
One nit, two Windows build fixes needed. Otherwise looks great!

jernoble committed:
Reduce build times by refactoring FrameView.h
https://bugs.webkit.org/show_bug.cgi?id=251626
rdar://104975520

Reviewed by Brent Fulgham.

Forward declare more things in FrameView.h and remove #include statements.

In order to forward-declare some enumerations used in FrameView.h, those enums
needed to be moved out of their classes' scopes, and in one case, need to be
converted to an OptionSet.

Canonical link: https://commits.webkit.org/259960@main
